src/project/internal/projectactionscontroller.cpp

```diff
@@ -64,8 +64,11 @@ void ProjectActionsController::init()
     dispatcher()->reg(this, "file-open", this, &ProjectActionsController::openProject);
 
     dispatcher()->reg(this, "file-close", [this]() {
-        bool quitApp = multiInstancesProvider()->instances().size() > 1;
-        closeOpenedProject(quitApp);
+        auto anyInstanceWithoutProject = multiInstancesProvider()->isHasAppInstanceWithoutProject();
+        closeOpenedProject(anyInstanceWithoutProject);
+        if (anyInstanceWithoutProject) {
+            multiInstancesProvider()->activateWindowWithoutProject();
```

Comment:
One thing to confirm, which is that "isHasAppInstanceWithProject()" quite possibly returns true if there's, for instance, a version of the process running that's doing some sort of command-line driven batch processing (e.g. audio export, assuming that's possible). In that case it should really be fixed not to do so, as it's not a "usable" app instance that you can usefully switch to.